client/checkout/blocks/confirm-upe-payment.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Handles the confirmation of card payments (3DSv2 modals/SCA challenge).
+ *
+ * @param {WCPayAPI} api            The API used for connection both with the server and Stripe.
+ * @param {string}   redirectUrl    The URL to redirect to after confirming the intent on Stripe.
+ * @param {Object}   paymentElement Reference to the UPE element mounted on the page.
+ * @param {Object}   billingData    An object containing the customer's billing data.
+ * @param {Object}   emitResponse   Various helpers for usage with observer response objects.
+ * @return {Object}                An object, which contains the result from the action.
+ */
+export default async function confirmUPEPayment(
+	api,
+	redirectUrl,
+	paymentElement,
+	billingData,
+	emitResponse
+) {
+	const name = (
+		billingData.first_name +
+		' ' +
+		billingData.last_name
+	).trim();
```

Comment:
```suggestion
	const name = `${ fields.billing_first_name } ${ fields.billing_last_name }`.trim();
```
Originally suggested by @frosso when I implemented this on the regular checkout, but I agree that this interpolation looks a little cleaner.